constellations/gemini-admin/src/user/user.js

```diff
@@ -0,0 +1,157 @@
+"use strict";
+require("./user.sass");
+import React from "react";
+import { Link } from "react-router-dom";
+import { debounce } from "lodash";
+import API from "../api.js";
+import { getCurrentUserSearch, setCurrentUserSearch } from "../localStorage.js";
+import DotsVertical from "../../assets/dots_vertical_gray.svg";
+
+export class UserPage extends React.Component {
+    state = {
+        list: [],
+        searchQuery: getCurrentUserSearch() || "",
+        currentDropdown: -1,
+    };
+
+    componentDidMount = () => {
+        API.post("api/users/search", { query: this.state.searchQuery }).then(
+            (res) => {
+                const users = res.data;
+                this.setState({ list: users });
+            }
+        );
+    };
+
+    searchUsers = debounce((query) => {
+        API.post("api/users/search", { query: query }).then((res) => {
+            const users = res.data;
+            this.setState({ list: users });
+        });
+    }, 200);
+
+    onChangeSearch = (event) => {
+        const query = event.target.value;
+
+        setCurrentUserSearch(query);
+        this.setState({ searchQuery: query }, () => {
+            this.searchUsers(this.state.searchQuery);
+        });
+    };
+
+    onClickDropdown = (id) => {
+        if (id == this.state.currentDropdown) {
+            this.setState({ currentDropdown: -1 });
+        } else {
+            this.setState({ currentDropdown: id });
+        }
+    };
+
+    render = () => {
+        const rows = this.state.list.map((row, index) => {
+            return (
+                <UserRow
+                    row={row}
+                    key={index}
+                    dropdownClickCallback={this.onClickDropdown}
+                    currentDropdown={this.state.currentDropdown}
+                />
+            );
+        });
+
+        return (
+            <div id="view-user">
+                <h1>Search Users</h1>
+
+                <input
+                    id="searchbar"
+                    value={this.state.searchQuery}
+                    onChange={this.onChangeSearch}
+                    placeholder="Search for a User"
+                />
+
+                <ul id="header">
+                    <li className="li-small">User ID</li>
+                    <li className="li-small">Account ID</li>
+                    <li className="li-med">Full Name</li>
+                    <li className="li-med">Email</li>
+                    <li className="li-med">Phone</li>
+                    <li className="li-large">Notes</li>
+                </ul>
+                {rows}
+            </div>
+        );
+    };
+}
+
+class UserRow extends React.Component {
+    render = () => {
+        const row = this.props.row;
+        var fullName = row.firstName;
+        if (row.middleName) {
+            fullName += " " + row.middleName + " " + row.lastName;
+        } else {
+            fullName += " " + row.lastName;
+        }
```

Comment:
did we ever get to creating a utils.createFullName(user)?

Reply:
oh right I missed adding it here